Static/DataSource.swift

```diff
@@ -204,13 +204,7 @@ extension DataSource: UITableViewDataSource {
     }
 
     public func tableView(_ tableView: UITableView, heightForHeaderInSection sectionIndex: Int) -> CGFloat {
-        guard let headerView = section(at: sectionIndex)?.header else { return 0 }
-
-        var headerHeight = headerView.viewHeight
-        if headerHeight == 0 {
-            headerHeight = UITableViewAutomaticDimension
-        }
-        return headerHeight
+        return section(at: sectionIndex)?.header?.viewHeight ?? UITableViewAutomaticDimension
```

Comment:
i think the thing we lose here is that if there's *no header* defined for this section, we're returning 'automatic dimension' now, versus in the previous code, we're returning `0` in that case.

Reply:
maybe this returns `0` if the table style is 'plain', and auto dimension if the table style is 'grouped' ??

Reply:
Double checking plain vs grouped once more